### PR TITLE
machinectl: add --now to start/stop containers when enabling/disabling

### LINKED ISSUE #26154: machinectl enable --now machinename
### Component

_machinectl_

### Is your feature request related to a problem? Please describe

Please, add `--now` parameter to `machinectl enable` - similar to systemctl.
The combined enable and start operations are often used.
In addition, we refer to the systemctl syntax.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### The systemd version you checked that didn't have the feature you are asking for

_No response_

### PR DESCRIPTION
**Pull request by YHNdnzj**
Closes #26154

### CODE REVIEW

File: src/machine/machinectl.c [hunk 1]
```diff
@@ -1667,7 +1668,21 @@ static int enable_machine(int argc, char *argv[], void *userdata) {
                 goto finish;
         }
 
-        r = 0;
+        if (arg_now) {
+                _cleanup_strv_free_ char **new_args = NULL;
+
+                new_args = strv_new(streq(argv[0], "enable") ? "start" : "stop");
+                if (!new_args) {
+                        r = -ENOMEM;
```
Comment (poettering) [inline src/machine/machinectl.c]:
```c
r = log_oom()
```

i.e. we should log about all errors here, and given nothing logged here so far we should do so here

File: src/machine/machinectl.c [hunk 2]
```diff
@@ -1667,7 +1668,21 @@ static int enable_machine(int argc, char *argv[], void *userdata) {
...
+                        goto finish;
+                }
+
+                r = strv_extend_strv(&new_args, argv + 1, /* filter_duplicates = */ false);
+                if (r < 0)
+                        goto finish;
```
Comment (poettering) [inline src/machine/machinectl.c:1683]: also log_oom() here.

(strv_extend_strv() can only fail on ENOMEM, hence we can sloppily just say:

```c
r = strv_extend_strv(…)
if (r < 0) {
        log_oom();
        goto finish;
}

**Comment by poettering** [comment]
looks good, just some log messages missing

File: src/machine/machinectl.c [hunk 3]
```diff
@@ -1616,7 +1617,7 @@ static int enable_machine(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return bus_log_create_error(r);
 
-        for (int i = 1; i < argc; i++) {
```
Comment (bluca) [inline src/machine/machinectl.c]: why is this being moved? we do want to declare variables in the closest scope possible

**Review by bluca — APPROVED** [review]

**Pull request merged**